src/main/java/edu/cornell/kfs/tax/batch/service/impl/TaxFileRowWriterImpl.java

```diff
@@ -159,7 +159,14 @@ private String getNormalizedStringValue(final TaxOutputFieldV2 field, final Bean
     }
 
     private String getPotentiallyMaskedValue(final TaxOutputFieldV2 field, final BeanWrapper wrappedDto) {
-        return maskSensitiveData ? field.getMask() : getStringValue(field, wrappedDto);
+        final String value = getStringValue(field, wrappedDto);
+        if (StringUtils.isBlank(value)) {
+            return KFSConstants.EMPTY_STRING;
+        } else if (maskSensitiveData) {
+            return field.getMask();
```

Comment:
should if (maskSensitiveData) be the first check?  That would mean we always shows the masked value when we are in a masking configuration

Reply:
I guess it depends on what the intended behavior should be if the value in question is already null/blank. Should the mask be printed anyway in such cases, or would it be better to keep it empty?

The old code might be printing empty strings for empty sensitive fields, but we could have the new process always print the masked value to the files if you think that would be better.

Reply:
If the old code is using empty string first, then we should probably leave the code as is.  If you think it is worth while, we could ask the functionals if they have a preference.